### DISCUSSION TITLE
Record plugin on Safari/iOS

### DISCUSSION THREAD

**Discussion by nizar** [katspaugh/wavesurfer.js#3759] [Q&A]
Hi, thanks for putting together this awesome package.

One quick question/issue. It appears that audio generated by the record plugin on Safari and iOS cannot be properly processed by the OpenAI whisper model. This is a known issue that is easily addressed by passing a `timeslice` param of 1000 to the `mediaRecorder.start()` call. I added this in a [fork](https://github.com/nizar/wavesurfer.js) and am happy to put in a PR.

Has anyone found an alternative solution?

Thanks

**Comment by katspaugh** [comment]
Sure, go ahead. What does this param do?

**Reply by nizar** [reply]
I'm not an expert on this by any stretch so I'm relying on the [MDN description](https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder/start) which says that it limits the amount of data returned in each blob. For some reason, webkit based browsers seem to want a cap of 1000 ms on individual chunks.

There are references to this in the OpenAI discussion forums [here](https://community.openai.com/t/whisper-problem-with-audio-mp4-blobs-from-safari/322252).

I'll send the PR and happy to make adjustments.

**Reply by katspaugh** [reply]
Merged the PR, thank you. I’ll let you know when it’s released.
